### ISSUE TITLE
 error when searching for TSIDs by metricIDs in the previous indexdb: EOF

### ISSUE THREAD

**Issue by ebusto** [VictoriaMetrics/VictoriaMetrics#3515]
**Describe the bug**
After upgrading all components in our development VictoriaMetrics cluster from 1.85.1 to 1.85.2, a subset of queries return an error. Caches were cleared by touching `reset_cache_on_startup` after upgrading.

```
422: error occured during search: cannot fetch query results from vmstorage nodes: cannot perform search on vmstorage dc6-itmp-dev-02.nvidia.com:8401: cannot execute funcName="search_v7" on vmstorage "10.128.124.201:8401": error when searching for tagFilters=[{AccountID=0,ProjectID=0,__name__="node_cpu_seconds_total",instance=~".+",mode="idle"}] on the time range [2022-12-20T19:30:00Z..2022-12-20T20:36:00Z]: error when searching for TSIDs by metricIDs in the previous indexdb: EOF
```

The query issued by Grafana:
```
topk_avg(5, 100 - (avg by (tenant) (irate(node_cpu_seconds_total{mode="idle", platform=~".*", service=~".*", role=~".*", site=~".*", status=~".*", tenant=~".*", instance=~".+"}) * 100)))
```

When scoping a label, such as `tenant`, to a single value, the query is successful. Similar queries on the same dashboard, such as the following, do not encounter an error.

```
topk_avg(5, avg by (tenant) (node_load1{platform=~".*", service=~".*", role=~".*", site=~".*", status=~".*", tenant=~".*", instance=~".+"}))
```

Downgrading vmstorage to 1.85.1 eliminates the error, and all queries are successful again.

**Logs**
Nothing at all, but we run vmstorage with `-loggerLevel=WARN`.

**Version**
We use the `victoriametrics/vmstorage:v1.85.2-cluster` Docker image.

**Arguments**
```
-dedup.minScrapeInterval=60s
-loggerLevel=WARN
-memory.allowedPercent=100
-retentionPeriod=7d
-storageDataPath=/data
```

**Comment by valyala** [comment]
@ebusto , thanks for reporting the bug! It should be fixed in the commit 1ff62629f46936cd7b664868067c7e61bab5a173 . Could you try upgrading VictoriaMetrics cluster components to the Docker image tag `heads-public-cluster-0-g1ff62629f` and check whether this fixes the error? Alternatively, you can build `vmstorage` from the commit mentioned above according to [these docs](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#building-from-sources), update all the `vmstorage` nodes to the newly built Docker image and verify whether this fixes the issue?

**Comment by ebusto** [comment]
@valyala Has anyone told you how awesome you are lately?

I rolled out your fix and everything looks great now. Thanks for the quick response.

**Comment by valyala** [comment]
@ebusto , thank for the confirmation that the bug has been fixed! The bugfix has been included in [v1.85.3 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases)
